osu.Game/GameModes/Menu/ButtonSystem.cs

```diff
@@ -264,6 +264,9 @@ protected override void Update()
         {
             //if (OsuGame.IdleTime > 6000 && State != MenuState.Exit)
             //    State = MenuState.Initial;
+            
+            if (state == MenuState.Exit && osuLogo.Alpha <= 0)                        
+                System.Environment.Exit(1);            

```

Comment:
You will need to add to add it to Game inside the framework, but it would just call `host.Exit()`.